### PR TITLE
AA-453: Implement initial EIP-7702 UserOperation test

### PR DESCRIPTION
**Pull request by forshtat**

### CODE REVIEW

File: tests/eip7702/test_eip7702_tuple_userop.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+    # create an EIP-7702 authorization tuple
+    nonce = w3.eth.get_transaction_count(acc.address)
+    auth_tuple = TupleEIP7702(
+        chainId=hex(1337), address=impl7702.address, nonce=hex(nonce)
+    )
+    auth_tuple.sign(acc._private_key.hex())
+
+    userop.sender = acc.address
+    userop.authorizationList = [auth_tuple]
```
Comment (andysim3d) [inline tests/eip7702/test_eip7702_tuple_userop.py:38]: just curious - for a single user operation, is that possible to attach multiple authorization tuples? and if so, what would be the use case? 

https://hackmd.io/WJtBdwvnT5yLq90F7UilIQ is pointing a single authorization tuples UO structure which I personally think it makes lot of sense.

File: tests/user_operation_erc4337.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    verificationGasLimit: HexStr = hex(10**6)
+    preVerificationGas: HexStr = hex(3 * 10**5)
+    maxFeePerGas: HexStr = hex(4 * 10**9)
+    maxPriorityFeePerGas: HexStr = hex(3 * 10**9)
+    signature: HexStr = "0x"
+    paymaster: HexStr = None
+    paymasterData: HexStr = None
+    paymasterVerificationGasLimit: HexStr = None
+    paymasterPostOpGasLimit: HexStr = None
+    authorizationList: list[TupleEIP7702] = None
```
Comment (andysim3d) [inline tests/user_operation_erc4337.py:28]: just curious - for a single user operation, is that possible to attach multiple authorization tuples? and if so, what would be the use case?

https://hackmd.io/WJtBdwvnT5yLq90F7UilIQ is pointing a single authorization tuples UO structure which I personally think it makes lot of sense.

Comment (shahafn) [reply tests/user_operation_erc4337.py:28]: Hey @andysim3d we're currently working on the validation rules, including whether we should allow more than one tuple. There are some use cases for that, but also some issues. We'll update soon

File: tests/eip7702/test_eip7702_tuple_userop.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    global last_account
+    last_account = acc
+    # delegated EOA account can actually have a state
+    state_after = eoa_with_authorization.functions.state().call()
+    assert state_after == 1111111
+
+
+# #normal transaction, using the same sender
+# TODO: must follow previous test, which deploys this account
```
Comment (shahafn) [inline tests/eip7702/test_eip7702_tuple_userop.py:69]: Please have them standalone

File: tests/eip7702/test_eip7702_tuple_userop.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    send_bundle_now,
+)
+
+AUTHORIZED_ACCOUNT_PREFIX = "ef0100"
+
+last_account = None
+
+
+@pytest.fixture
+def impl7702(w3):
```
Comment (shahafn) [inline tests/eip7702/test_eip7702_tuple_userop.py:19]: Please define fixtures in conftest.py

File: tests/eip7702/test_eip7702_tuple_userop.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    return deploy_contract(w3, "SimpleWallet", ctrparams=[CommandLineArgs.entrypoint])
+
+
+def test_send_eip_7702_tx(w3, userop, impl7702, wallet_contract, helper_contract):
```
Comment (shahafn) [inline tests/eip7702/test_eip7702_tuple_userop.py:23]: What about more than one tuple?

File: tests/eip7702/test_eip7702_tuple_userop.py (hunk 4 above)
Comment (forshtat) [reply tests/eip7702/test_eip7702_tuple_userop.py]: Fixed.

File: .circleci/config.yml [hunk 6]
```diff
@@ -132,6 +131,7 @@ jobs:
           name: "Run go-ethereum"
           working_directory: "./go-ethereum"
           command: "\
+          ./build/bin/geth version ; \
```
Comment (shahafn) [inline .circleci/config.yml:134]: Please remove

File: .circleci/config.yml [hunk 7]
```diff
@@ -286,6 +287,84 @@ jobs:
...
+      - run:
+          name: "curl pdm"
+          command: "curl -sSL https://raw.githubusercontent.com/pdm-project/pdm/main/install-pdm.py | python3 -"
+      - run:
+          name: "update-deps"
+          command: "pdm run update-deps"
+      - run:
+          name: "pdm install"
+          command: "pdm install"
+      # - run:
```
Comment (shahafn) [inline .circleci/config.yml:331]: Please remove

**Review by shahafn — APPROVED** [review]

**Pull request merged**